### DISCUSSION TITLE
Store translated Content

### DISCUSSION THREAD

**Discussion by johannesschobel** [prisma/prisma#5710] [Q&A]
Dear all,

in my current application scenario, i need to store content in various languages. The user interacting with the API, in turn, can select respective content-language via `http-headers`.

Consider the following example:
A `Course` model may have the `name` and `description` attributes, that describe this particular resource. This information should be provided in various languages, like `en` (English), `de` (German) or `fr` (French). A user then requests a particular language via
```
GET /courses/1
Accept-Language: de
```

The service will read the respective `Course` with `id = 1` and will output the localized information based on the request.

How shall we store this language-dependent information properly?
For now, i see 2 different approaches:

## Approach A: Localization in a dedicated Table

You would need to have a `Course` and `CourseTranslation` table, whereas the `CourseTranslation` would hold translatable information (`name` and `description`) with respective language tag. You would need to join information before returning it to the client. This approach basically doubles the amount of tables you'll have.

## Approach B: Store Content as JSON

You would need to change the `name` and `description` from type `String` to `Json`. Then you could just store data like this: 
```json
{
  "de": "Deutscher Titel",
  "en": "English Title",
  "fr": "actually i don't speak French but you will get the idea ;)"
}
```

I guess, Approach B, is easier to maintain, however, you store a lot (!) of additional information. Furthermore, querying data with patterns like _"get all courses that contain "healthcare"_ are way (!) more (!) complicated than with Approach A.

Are there additional approaches? What do you Prisma guys think of this scenario? What are your best practices?

All the best

**Comment by ryands17** [comment]
@johannesschobel 👋 

I would prefer storing it using **Approach B** as you can store the course values in other fields. Also, querying data like _get all courses that contain "healthcare"_ are possible as you can index based on the key in your Json field. Postgres supports that and it will be quite quick due to such indexes.

This would be much better than storing it in a dedicated table as this data seems better suited to `Json`.

**Reply by johannesschobel** [reply]
i agree.. I would also vote for `Approach B` as it is also a better maintainable approach, i think.
Does the `where` clause from prisma support the indexed json search from postgres? Or is it basically just a `where name like '%healthcare%'` search string that is applied?

All the best

**Reply by ryands17** [reply]
Currently Prisma doesn't support the indexed Json search, so you would need to use a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access). But we have a [feature request](https://github.com/prisma/prisma/issues/2444) for the same so it would be great if you could add a 👍 to this so that we can set the priority :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
